### PR TITLE
feat: A method for getting unassigned IPs

### PR DESCRIPTION
**Pull request by petmat**
I think this should implement the method that was described in issue #569 . I added a test to see that the IPs without subusers are returned and the ones with the subusers are not.

It's my first time contributing to this repo so please if there is something wrong with my PR let me know and I will get it fixed!

### CODE REVIEW

**Comment by SendGridDX** [comment]
[CLA assistant check](https://cla.sendgrid.com/sendgrid/sendgrid-csharp?pullRequest=573) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.sendgrid.com/sendgrid/sendgrid-csharp?pullRequest=573) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.sendgrid.com/check/sendgrid/sendgrid-csharp?pullRequest=573) it.</sub>

**Comment by thinkingserious** [comment]
@petmat

We have not been able to merge your Pull Request, but because you are awesome - we wanted to make sure you could still get a SendGrid Hacktoberfest shirt.  

Please go **fill [out our swag form](https://dx.sendgrid.com/swag) before Nov 5th** and we will send the shirt! (We know that you might have tried this before and it didn’t work, sorry about that!)

### **You have till Nov 5th to fill out this form** in order to get the Hacktoberfest shirt!

Thank you for contributing during Hacktoberfest! We hope to see you in the repos soon! Just so you know, we always give away a SendGrid shirt for your first ever non-Hacktoberfest PR that gets merged.

**Comment by thinkingserious** [comment]
Hello @petmat,

No worries, I'm grateful for your contribution!

Your code should go into it's own Helper directory and I think the name you chose works, with documentation added [here](https://github.com/sendgrid/sendgrid-csharp/blob/master/USE_CASES.md).

I do like the proposed interface `var ips = await sg.GetUnassignedIpsAsync();`, nice and simple. 

I'm definitely open to suggestions. I'm just trying to make sure as we add more of these helpers, we do so in a consistent, scalable manner. We have over 230+ other endpoints that we will be wanting to create helpers for.

With Best Regards,

Elmer

**Comment by petmat** [comment]
A made an effort to refactor the code like you suggested. Hopefully this i atleast closer to what was intended.

File: USE_CASES.md [hunk 1]
```diff
@@ -891,8 +895,40 @@ namespace Example {
...
+using System.Threading.Tasks;
+using System.IO;
+
+namespace Example
+{
+    internal class Example
+    {
+        private static void Main()
+        {
+            ExecuteManualAttachmentAdd().Wait();
```
Comment (thinkingserious) [inline USE_CASES.md]: I think this should be ExecuteGetUnassignedIps().Wait();

File: src/SendGrid/Helpers/IpAddresses/README.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+using System.Threading.Tasks;
+using System.IO;
+
+namespace Example
+{
+    internal class Example
+    {
+        private static void Main()
+        {
+            ExecuteManualAttachmentAdd().Wait();
```
Comment (thinkingserious) [inline src/SendGrid/Helpers/IpAddresses/README.md]: I think this should be ExecuteGetUnassignedIps().Wait();

File: src/SendGrid/Helpers/IpAddresses/IpAddressHelper.cs [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        /// </summary>
+        /// <returns>A list of IP addresses</returns>
+        public async Task<IList<string>> GetUnassignedIpsAsync()
+        {
+            var response = await this.sendGridClient.RequestAsync(
+                Method.GET,
+                null,
+                urlPath: "ips");
+
+            dynamic result = JArray.Parse(await response.Body.ReadAsStringAsync());
```
Comment (thinkingserious) [inline src/SendGrid/Helpers/IpAddresses/IpAddressHelper.cs]: Why a dynamic type here?

File: src/SendGrid/Helpers/IpAddresses/IpAddressHelper.cs [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+            return ((IEnumerable<dynamic>)result)
```
Comment (thinkingserious) [inline src/SendGrid/Helpers/IpAddresses/IpAddressHelper.cs]: Why the dynamic type here and below?

**Comment by thinkingserious** [comment]
Hello @petmat,

Just checking in. Please let me know what you think about my requested changes. Thanks!

**Comment by thinkingserious** [comment]
Closing until PR feedback is addressed.

**Pull request closed without merging**